mesonbuild/mintro.py

```diff
@@ -228,74 +204,72 @@ def list_buildoptions(coredata: cdata.CoreData) -> List[dict]:
     test_options = {k: o for k, o in coredata.builtins.items() if k in test_option_names}
     core_options = {k: o for k, o in coredata.builtins.items() if k in core_option_names}
 
-    add_keys(optlist, core_options, 'core')
-    add_keys(optlist, coredata.builtins_per_machine.host, 'core', machine='host')
+    def add_keys(options: Dict[str, cdata.UserOption], section: str, machine: str = 'any') -> None:
+        keys = list(options.keys())
+        keys.sort()
```

Comment:
please don't use container.sort() if you're not going to retain the sorted values. That does an in place sort and is really expensive,
```python
for key in sorted(options.keys()):
    ...
```

is what you really want here.

Reply:
Right, I missed that in the refactor. Fixed, thanks.